nav2_map_server/include/nav2_map_server/vector_object_shapes.hpp

```diff
@@ -0,0 +1,456 @@
+// Copyright (c) 2023 Samsung R&D Institute Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_MAP_SERVER__VECTOR_OBJECT_SHAPES_HPP_
+#define NAV2_MAP_SERVER__VECTOR_OBJECT_SHAPES_HPP_
+
+#include <cmath>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "geometry_msgs/msg/polygon.hpp"
+#include "geometry_msgs/msg/point32.hpp"
+#include "nav_msgs/msg/occupancy_grid.hpp"
+
+#include "tf2_ros/buffer.h"
+
+#include "nav2_msgs/msg/polygon_vo.hpp"
+#include "nav2_msgs/msg/circle_vo.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+
+#include "nav2_map_server/vector_object_utils.hpp"
+
+namespace nav2_map_server
+{
+
+/// @brief Possible VO-shape types
+enum ShapeType
+{
+  UNKNOWN = 0,
+  POLYGON = 1,
+  CIRCLE = 2
+};
+
+/// @brief Basic class, other vector objects to be inherited from
+class Shape
+{
+public:
+  /**
+   * @brief Shape basic class constructor
+   * @param node Vector Object server node pointer
+   */
+  explicit Shape(const nav2_util::LifecycleNode::WeakPtr & node);
+
+  /**
+   * @brief Shape destructor
+   */
+  virtual ~Shape();
+
+  /**
+   * @brief Returns type of the shape
+   * @return Type of the shape
+   */
+  ShapeType getType();
+
+  /**
+   * @brief Supporting routine obtaining shape UUID from ROS-parameters
+   * for the given shape object
+   * @return True if UUID was obtained or false in failure case
+   */
+  bool getShapeUUID(const std::string & shape_name, unsigned char * out_uuid);
+
+  /**
+   * @brief Supporting routine obtaining ROS-parameters for the given vector object.
+   * Empty virtual method intended to be used in child implementations
+   * @param shape_name Name of the shape
+   * @return True if all parameters were obtained or false in failure case
+   */
+  virtual bool getROSParameters(const std::string & shape_name) = 0;
```

Comment:
All methods of this name: all the parameters are "ROS" parameters, so that is redundant